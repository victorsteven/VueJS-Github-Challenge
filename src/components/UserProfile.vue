<template>
  <div>
     <article class="panel">
        <div class="panel_right">
          <h2 class="panel_name">{{ user.name }}</h2>
          <div class="panel_bio">{{ user.bio }}</div>
          <div class="panel_followers" v-if="user.followers">Followers: {{ user.followers }}</div>
          <p>{{ user.location }}</p>
        </div>
        <div class="panel_left">
          <div class="panel_image">
            <!-- <img v-if="user.avatar_url" :src="user.avatar_url" :alt="user.name"> -->
            <img class="panel_image_avatar" v-if="user.avatar_url" :src="user.avatar_url" :alt="user.name">

          </div>
        </div>
      </article>
  </div>
</template>

<script>
export default {
  props: {
    user: {
      type: Object,
      required: true,
      default: () => ({}),
    },
  },
};
</script>
