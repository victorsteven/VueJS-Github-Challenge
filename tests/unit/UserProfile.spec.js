import { shallowMount } from '@vue/test-utils';
import UserProfile from '../../src/components/UserProfile.vue';
import user from '../mockData';


describe('UserProfile', () => {
  let props;

  const util = () => {
    const wrapper = shallowMount(UserProfile, {
      propsData: props,
    });
    return {
      wrapper,
      avatar: () => wrapper.find('.panel_image_avatar'),
      name: () => wrapper.find('.panel_name'),
      bio: () => wrapper.find('.panel_bio'),
    };
  };

  beforeEach(() => {
    props = {
      user,
    };
  });

  it('renders the UserProfile component', () => {
    const { wrapper } = util();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders UserProfile Data components', () => {
    const { avatar, name, bio } = util();
    expect(avatar().exists()).toBe(true);
    expect(avatar().attributes().src).toBe(props.user.avatar_url);

    expect(name().exists()).toBe(true);
    expect(name().text()).toBe(props.user.name);

    expect(bio().exists()).toBe(true);
    expect(bio().text()).toBe(props.user.bio);
  });
});
