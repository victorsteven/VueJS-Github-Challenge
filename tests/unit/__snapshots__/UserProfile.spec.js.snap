// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile renders the UserProfile component 1`] = `
<div>
  <article class="panel">
    <div class="panel_right">
      <h2 class="panel_name">Steven Victor</h2>
      <div class="panel_bio">Web Developer</div>
      <div class="panel_followers">Followers: 20</div>
      <p>Nigeria</p>
    </div>
    <div class="panel_left">
      <div class="panel_image"><img src="https://github.com/images/error/victorsteven_happy.gif" alt="Steven Victor" class="panel_image_avatar"></div>
    </div>
  </article>
</div>
`;

exports[`UserProfile renders the component 1`] = `
<div>
  <article class="panel">
    <div class="panel_right">
      <h2 class="panel_name">Steven Victor</h2>
      <div class="panel_bio">Web Developer</div>
      <div class="panel_followers">Followers: 20</div>
      <p>Nigeria</p>
    </div>
    <div class="panel_left">
      <div class="panel_image"><img src="https://github.com/images/error/victorsteven_happy.gif" alt="Steven Victor" class="panel_image_avatar"></div>
    </div>
  </article>
</div>
`;
